packages/presentation/lib/main_page.dart

```diff
@@ -0,0 +1,146 @@
+import 'package:domain/usecase/check_number_usecase.dart';
+import 'package:domain/usecase/generate_number_usecase.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'main_bloc.dart';
+import 'main_event.dart';
+import 'main_state.dart';
+
+class MyApp extends StatefulWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<MainBloc>(
+      create: (_) => MainBloc(
+        checkNumberUseCase: CheckNumberUseCase(),
+        generateNumberUseCase: GenerateNumberUsecase(),
+      ),
+      child: MaterialApp(
+          title: 'Flutter Demo',
+          debugShowCheckedModeBanner: false,
+          theme: ThemeData(
+            primarySwatch: Colors.deepPurple,
+          ),
+          home: const MyHomePage()),
+    );
+  }
+}
+
+class MyHomePage extends StatelessWidget {
+  const MyHomePage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final model = context.watch<MainBloc>();
```

Comment:
Seems that the same model can be obtained from builder: (context) on 86 line, which prevents 39 line unnecessary  rebuilds.

Reply:
fixed in fix move model to builder bloc consumer commit